components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -66,34 +136,35 @@ class Experiments(
      * the specified experiment
      *
      * @param context context
-     * @param descriptor descriptor of the experiment to check
+     * @param experimentId the id of the experiment
      *
      * @return true if the user is part of the specified experiment, false otherwise
      */
-    fun isInExperiment(context: Context, descriptor: ExperimentDescriptor): Boolean {
-        return evaluator.evaluate(context, descriptor, experimentsResult.experiments) != null
+    fun isInExperiment(context: Context, experimentId: String): Boolean {
+        return evaluator.evaluate(context, ExperimentDescriptor(experimentId), experimentsResult.experiments) != null
     }
 
     /**
      * Performs an action if the user is part of the specified experiment
      *
      * @param context context
-     * @param descriptor descriptor of the experiment to check
+     * @param experimentId the id of the experiment
      * @param block block of code to be executed if the user is part of the experiment
      */
-    fun withExperiment(context: Context, descriptor: ExperimentDescriptor, block: (Experiment) -> Unit) {
-        evaluator.evaluate(context, descriptor, experimentsResult.experiments)?.let { block(it) }
+    fun withExperiment(context: Context, experimentId: String, block: () -> Unit) {
+        evaluator.evaluate(context, ExperimentDescriptor(experimentId), experimentsResult.experiments)?.let { block() }
```

Comment:
Should we run this `block()` in it's own thread?  I mean, this could potentially do IO or some other time/resource intensive thing that we might not want to bog down the current thread.

Reply:
That's an interesting question.
I'd say this is for the product engineers to decide.
I assume this needs to behave synchronously by default, so, as a product engineer, i can write code as if it was just a logic branch in my function.
If async is needed, they can decide by themselves.

Reply:
That makes sense... we expect to execute this synchronously, so any async requirements would be the responsibility of the caller and handled withing the `block` closure.  That answers my question.